### DISCUSSION TITLE
Labels: add filtering by "no labels" (similarly - for "area" & "blueprint")

### DISCUSSION THREAD

**Discussion by ildar170975** [home-assistant/frontend#20418] [Dashboard]
Currently it is possible to filter a table by some label (labels):

![image](https://github.com/home-assistant/frontend/assets/71872483/5534b218-cfed-4392-9fcf-572e4e00a012)

Suggest to add one more criteria - "no labels" (or with a similar meaning):

![image](https://github.com/home-assistant/frontend/assets/71872483/397108b5-9ab0-4a96-8ea8-005b86a582c4)

If this criteria is selected - a table will contain elements w/o labels associated.
This criteria cannot be selected together with other label: if "no labels" is selected - then other label-based criteria are ignored.

-------

Similarly:
- for "Areas" - to allow select entities which are not associated to any area;
- for "Blueprint" - to allow select automations/scripts which are not based on any blueprints.

**Comment by mrgrlscz** [comment]
Yes, should be a great thing to add a "no label" AND a "no category" filter.
But you have to be able to check it together with other label /category. This way you could display i.e. every automation, except the ones wich labels are not selected.

**Reply by ildar170975** [reply]
Probably you are right. It will be more flexible.

**Comment by MikeGuest** [comment]
+1

**Comment by ildar170975** [comment]
And suggest to add this "No labels" (or "None") on the TOP of the list.
Example - list of blueprints can be huge, so it is easier to select the "None" checkbox if it is on the top:

![image](https://github.com/home-assistant/frontend/assets/71872483/30057e77-8f7a-4710-afb1-f1bd4fccb90a)

**Comment by ildar170975** [comment]
Same about categories - https://community.home-assistant.io/t/wth-why-cant-i-filter-to-a-uncategorized-automations-etc/805659

**Comment by rtvanhook** [comment]
Yes!  Same with every list that can be filtered.  As someone would like to identify items in a filtered list that are missing whatever the filter needs, I would like to be able to filter for items that can't be filtered.

**Comment by prorun26** [comment]
This feature would be highly beneficial for identifying and organizing unclassified entities more efficiently.

**Use Case**
Allows users to quickly find entities that have not been assigned labels.
Particularly useful for discovering new entities, especially in large Home Assistant setups.

**Implementation Suggestion**
Extend the existing label filter logic by introducing a 'No Label' option.
Initially, I considered that selecting this option should automatically deselect all other label filters to ensure that only unlabeled entities are displayed. However, an even more flexible approach would be to allow label filters to be combined. For example, enabling users to filter for both unlabeled entities and entities with a specific label, such as "Show all entities without a label AND all entities with label XYZ."

**Comment by guyhindle** [comment]
Also... `NOT <label>`

**Comment by Hakun1n** [comment]
Yes please!

**Comment by ma-i0** [comment]
+1

**Comment by ildar170975** [comment]
PR for adding a "none of listed" variant for "Devices" - https://github.com/home-assistant/frontend/issues/29550
Similar PRs for labels, assistants, floors/areas, blueprints will be made if the 1st PR is accepted in general.
Categories will be considered later as well.
